src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -49,7 +60,7 @@ public static partial class JsonSerializer
 
                     if (createExtensionProperty)
                     {
-                        CreateDataExtensionProperty(obj, dataExtProperty, options);
+                        CreateDataExtensionProperty(obj!, dataExtProperty, options);
```

Comment:
Under what circumstances could this be null? Could you add a Debug.Assert to catch any potential bugs?